ParsingNotes.md

```diff
@@ -0,0 +1,167 @@
+## Parsing
+
+There are general steps KaTeX takes when processing an input into
+HTML and MathML.
+
+1. Take input string and convert to a "tree".  This is somewhat of an
+   intermediate abstract syntex tree and has the form of a `Array(ParseNode)`.
+   These will be described below.
+2. Take the `Array(ParseNode)` and convert to MathML.
+3. Take the `Array(ParseNode)` and convert to HTML.
+
+There is one somewhat important intermediate step taking in steps 2 and 3, which
+is that KaTeX first takes this array of `ParseNode`s and converts it into
+a an object wich represents the tree of markup that will be converted into actual
+markup.  The reason this is somewhat of an important intermediate step is that this
+"markup tree" found in the middle of step 3 is the closest thing that you will find
+to an actual TeX abstract syntex tree.  Mostly since both TeX and HTML both like to
+think of everything as boxes and how these boxes should interact with eachother.
+
+We will first spend some time trying to understand this first step.  The intermediate
+syntax tree which is simply an array of `ParseNode`s.
+
+## ParseNodes
+
+The folowing is a list of currently implemented ParseNodes.  They can roughly
+categorized into four different groups.
+
+1. [**Symbols**](#Symbols)
+2. [**Math Lists**](#Math-Lists)
+3. [**Functions**](#Functions)
+4. [**Wrappers**](#Wrappers)
+
+### Symbols
+
+Syombols are the easiest of the four groups in that they have a minimal structure.
+A typical example is the following:
+
+```
+type: "bin"
+value: "+"
+mode: "math"
+```
+Any token that can be found in `symbols.js` will have this structure.  It can occur
+in either `math` mode or `text` mode, and the type can be any of the basic `atom`
+types defined in TeX--accent, bin, close, inner, mathord, op, open, punct, rel--but
+also include two special cases: spacing, textord.  The value is simply the unicode
+character corresponding to the given symbol, for example `\u222b` is the symbol
+corresponding to the integration operator `\int`; This is the replace field in the
+definition of a symbol, and is not included in the `ParseNode`. The file `symbols.js` 
+also contains information about which font the unicode symbol is placed in.  The
+general design rule for not including this information in the `ParseNode` is to not
+incorporate more infromation that is necessary for `buildMathML` to do it's job.
+
+### Math Lists
+
+Math lists are supposed to be like a new list of `ParseNodes`.  This allows for
+easy recursion while rendering in the same spirit that most reference TeX
+implementations will compile.  As of right now there is only on example, the
```

Comment:
The last part of the "This allows for... " sentence doesn't make sense.  Maybe split them into separate sentences.